internal/controllers/topology/cluster/reconcile_state.go

```diff
@@ -727,3 +728,26 @@ func (r *Reconciler) reconcileReferencedTemplate(ctx context.Context, in reconci
 
 	return nil
 }
+
+// createErrorWithoutObjectName removes the name of the object from the error message. As each new Create call involves an
+// object with a unique generated name each error appears to be a different error. As the errors are being surfaced in a condition
+// on the Cluster, the name is removed here to prevent each creation error from triggering a new reconciliation.
+func createErrorWithoutObjectName(err error, obj client.Object) error {
+	var statusError *apierrors.StatusError
+	if errors.As(err, &statusError) {
+		out := ""
+		if statusError.Status().Details != nil {
+			for _, cause := range statusError.Status().Details.Causes {
+				out = strings.Join([]string{out, fmt.Sprintf("%s: %s: %s", cause.Type, cause.Field, cause.Message)}, "") //nolint:gocritic
+			}
+			out = fmt.Sprintf("%s.%s: %s", statusError.Status().Details.Kind, statusError.Status().Details.Group, out)
+		}
+		if len(out) > 0 {
+			// Replace the statusError message with the constructed message.
+			statusError.ErrStatus.Message = fmt.Sprintf("failed to create %s", out)
+			return statusError
+		}
+	}
```

Comment:
```suggestion
	if errors.As(err, &statusError) {
		if statusError.Status().Details != nil {
			var causes []string
			for _, cause := range statusError.Status().Details.Causes {
				causes = append(causes, fmt.Sprintf("%s: %s: %s", cause.Type, cause.Field, cause.Message))
			}

			var msg string
			if len(causes) > 0 {
				msg = fmt.Sprintf("failed to create: %s.%s: %s", statusError.Status().Details.Kind, statusError.Status().Details.Group, strings.Join(causes, " "))
			} else {
				msg = fmt.Sprintf("failed to create: %s.%s", statusError.Status().Details.Kind, statusError.Status().Details.Group)
			}

			// Replace the statusError message with the constructed message.
			statusError.ErrStatus.Message = msg
			return statusError
		}
	}
```
I should have explained a bit more. I meant something like this with `strings.Join`. So strings.Join can solve the problem with leading and trailing slashes. 

I think this way we wouldn't end up with cases were we get awkward trailing spaces (like e.g. when we don't have any causes)